climpred/utils.py

```diff
@@ -68,6 +132,38 @@ def get_comparison_class(comparison, list_):
         return getattr(comparisons, '__' + comparison)
 
 
+def get_lead_cftime_shift_args(units, lead):
+    """Determines the date increment to use when adding the lead time to init time based
+    on the units attribute.
+
+    Args:
+        units (str): Units associated with the lead dimension. Must be
+            years, seasons, months, weeks, pentads, days.
+        lead (int): increment of lead being computed.
+
+    Returns:
+       offset_args_tuple (tuple): Tuple containing (value, str) for ``CFTime.shift()``.
+    """
+    lead = int(lead)
+    if units == 'years':
+        # Currently assumes yearly aligns with year start.
+        offset_args_tuple = (lead, 'YS')
+    elif units == 'months':
+        # Currently assume monthly aligns with month start.
+        offset_args_tuple = (lead, 'MS')
+    elif units == 'weeks':
+        offset_args_tuple = (lead * 7, 'D')
+    elif units == 'days':
+        offset_args_tuple = (lead, 'D')
+    elif units == 'seasons':
+        offset_args_tuple = (lead * 3, 'MS')
+    elif units == 'pentads':
+        offset_args_tuple = (lead * 5, 'D')
+    else:
+        raise ValueError(f'{units} is not a valid choice')
```

Comment:
Could we replace the if/elif statements with a dictionary? With a dictionary, the error message could also be more useful if it listed valid values:

```python
d = {'years': (lead, 'YS'), 'months' : (lead, 'MS'), 'weeks': (lead * 7, 'D'), 
        'days': (lead, 'D'), ....}

try:
  offset_args_tuple = d[units]
except KeyError:
   print((f'{units} is not a valid choice'))
   print(f'Accepted `units` values include: {d.keys()}')
```

Reply:
Great point, thanks!